mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/LegacyProxyOperation.java

```diff
@@ -529,9 +516,11 @@ public void handleRollback(OperationContext context, ModelNode operation) {
     },
     ;
 
-    private static ServiceController<?> getService(OperationContext context) throws OperationFailedException {
-        PathAddress proxyAddress = LegacyMetricOperationsRegistration.translateProxyPath(context);
+    static final String SESSION_DRAINING_COMPLETE = "session-draining-complete";
+
+    private static ModClusterServiceMBean getService(OperationContext context) throws OperationFailedException {
+        PathAddress proxyAddress = LegacyMetricOperationsRegistration.translateProxyPath(context, context.getCurrentAddress());
         ServiceName serviceName = ProxyConfigurationResourceDefinition.Capability.SERVICE.getServiceName(proxyAddress);
-        return context.getServiceRegistry(false).getService(serviceName);
+        return new ActiveServiceSupplier<ModClusterServiceMBean>(context.getServiceRegistry(false), serviceName).get();
```

Comment:
For a lot of the uses of this method, this should be context.getServiceRegistry(true).

It doesn't need to be for the getService(context) != null checks, but for the other ones if the handler is changing any state it should be 'true'.

{code}
     * <strong>Note:</strong> It is very important that the {@code modify} parameter accurately reflect whether the
     * caller intends to make any modifications to any object reachable, directly or indirectly, from the returned
     * {@link ServiceRegistry}. This includes modifying any {@link ServiceController},
     * {@link org.jboss.msc.service.Service}, {@link org.jboss.msc.value.Value} or any object reachable from a value.
     *
     * @param modify {@code true} if the operation may be modifying any object reachable directly or indirectly from
     *                           the returned {@link ServiceRegistry}, {@code false} otherwise
{code}

Reply:
Fixed (not sure what it is for though and I assume there would be other usages of this incorrectly passing false, assuming its modifications to the registry itself, not the object internal state).